src/mono/wasm/build/WasmApp.Native.targets

```diff
@@ -600,6 +600,10 @@
     <PropertyGroup>
       <!-- FIXME: do it once -->
       <_MonoAotCrossCompilerPath>@(MonoAotCrossCompiler->WithMetadataValue('RuntimeIdentifier','browser-wasm'))</_MonoAotCrossCompilerPath>
+
+      <!-- find the path with the assemblies, so we don't have to hardcode the tfm.
+           Cannot use System.Private.Corelib since that is in a different directory -->
+      <_SystemIOPath>@(RuntimePackAsset->WithMetadataValue('Filename', 'System.IO'))</_SystemIOPath>
```

Comment:
Why System.IO? System.IO is a pure shim assembly which might get trimmed out in certain scenarios.

Reply:
Trimmed from the runtime pack?

They are in different paths:
`./Microsoft.NETCore.App.Runtime.Mono.browser-wasm/8.0.0-dev/runtimes/browser-wasm/native/System.Private.CoreLib.dll`
`./Microsoft.NETCore.App.Runtime.Mono.browser-wasm/8.0.0-dev/runtimes/browser-wasm/lib/net8.0/System.IO.dll`

.. and we need the latter one.

Reply:
Looks like this fails for AOT/helix where we create a proxy project, and run only the wasm part of the build. And that doesn't have `RuntimePackAsset`, because none of the targets related to that were run.

I will try to populate RuntimePackAsset in the proxy project, but that will be bit of a hack.